lib/actions/local/configure.coffee

```diff
@@ -95,100 +67,65 @@ inquirerOptions = (data) -> [
 	}
 ]
 
+joinPath = (path) -> '/' + path.join('/')
+
+load = (target, type) ->
+	{ Reconfix } = require('reconfix-preview')
+	imagefs = require('resin-image-fs')
+
+	reconfix = new Reconfix({
+		read: (partition, path) ->
+			imagefs.readFile
+				image: target
+				partition: partition
+				path: joinPath(path)
+			.catch (err) ->
+				if err?.code == 'NOENT'
+					return Buffer.alloc(0)
+				else
+					throw err
+			.then (content) -> Buffer.from(content, 'utf-8')
+			.catch (err) -> {}
+		write: (partition, path, data) ->
+			imagefs.writeFile
+				image: target,
+				partition: partition,
+				path: joinPath(path)
+			, data.toString('utf-8')
+	})
+	readFileAsync(path.join(__dirname, '../../schemas/resin-os.json'))
+	.then (content) -> JSON.parse(content)
+	.then (json) ->
```

Comment:
Not critical, but I guess you could write this as:

```
.then(reconfix.loadSchema)
.return(reconfix)
```

Reply:
I think that looks better.